### ISSUE TITLE
[Java] Maven Javadoc Error while building the generated code with Java 11

### ISSUE THREAD

**Issue by Grandolf49** [swagger-api/swagger-codegen#10421]
### Description

I am using **Java 11.0.3** with **swagger-codegen-cli:3.0.18** and **[org.hidetake:gradle-swagger-generator-plugin:2.18.2](https://github.com/int128/gradle-swagger-generator-plugin)** to build Java Retofit2 based API Client on Ubuntu. The code gets generated without any issue. But when I try to build the generated code using **mvn clean package** I get this error.

```console
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  04:04 min
[INFO] Finished at: 2020-08-04T12:52:58+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:2.10.4:jar (attach-javadocs) on project client: MavenReportException: Error while generating Javadoc: 
[ERROR] Exit code: 1 - javadoc: error - The code being documented uses modules but the packages defined in http://docs.oracle.com/javase/8/docs/api/ are in the unnamed module.
[ERROR] 
[ERROR] Command line was: /usr/lib/jvm/java-11-openjdk-amd64/bin/javadoc @options @packages
[ERROR] 
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

After googling, it seems that this is a bug in JDK but following workarounds are possible.
- Switch Java version to Java 8, the generated code builds successfully.
- To use Java 11, manually edit the pom.xml and change the version of `org.apache.maven.plugins:maven-javadoc-plugin:2.10.4` to `3.2.0` to build it successfully. 

Since it is not feasible to change to java 8 programmatically just for building this, I'm expecting that the javadoc version is updated in the template files itself.

### Related issues/PRs

- https://github.com/swagger-api/swagger-codegen/pull/10041 - This PR is outdated but addresses the issue
- https://github.com/swagger-api/swagger-codegen/issues/8959 - This is similar but 2 years ago

### Suggest a fix/enhancement

Changing the version to `3.2.0` in the template file will solve this issue.

https://github.com/swagger-api/swagger-codegen/blob/4607a90d7b69463a0ae8fc94fac68fc95d80965e/modules/swagger-codegen/src/main/resources/Java/libraries/retrofit/pom.mustache#L149

**Comment by jharrorsted** [comment]
I got the same issue. I tried your fix, but afterwards my 'mvn clean package' still fails.

**Comment by Grandolf49** [comment]
@jharrorsted  what build error are you encountering?

**Comment by jharrorsted** [comment]
> 
> 
> @jharrorsted what build error are you encountering?

Here is an exerpt:

  TypeScriptInversifyClientOptionsTest.checkOptionsHelp » NoClassDefFound javax/...
  TypeScriptInversifyClientOptionsTest.checkOptionsProcessing » NoClassDefFound ...
  TypeScriptNodeClientOptionsTest.checkOptionsHelp » NoClassDefFound javax/annot...
  TypeScriptNodeClientOptionsTest.checkOptionsProcessing » NoClassDefFound javax...

Tests run: 706, Failures: 131, Errors: 0, Skipped: 3

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for swagger-codegen-project 2.4.16-SNAPSHOT:
[INFO]
[INFO] swagger-codegen-project ............................ SUCCESS [  1.262 s]
[INFO] swagger-codegen (core library) ..................... FAILURE [ 32.539 s]
[INFO] swagger-codegen (executable) ....................... SKIPPED
[INFO] swagger-codegen (maven-plugin) ..................... SKIPPED
[INFO] swagger-generator .................................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  34.302 s
[INFO] Finished at: 2020-08-05T14:53:33+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (test-testng) on project swagger-codegen: There are test failures. [ERROR]
[ERROR] Please refer to C:\tools\swagger-codegen\modules\swagger-codegen\target\surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :swagger-codegen

**Comment by jharrorsted** [comment]
I just realized that I am using java 14.0.2, and not "java 11" as suggested

**Comment by Grandolf49** [comment]
This is happening because javax/annotations is deprecated in Java 11+
You can set the **hideGenerationTimestamp**  to **true** in your config.json file and try building it again.

**Comment by Grandolf49** [comment]
This is happening because javax/annotations is discontinued in Java 11+
You can set the **hideGenerationTimestamp**  to **true** in your config.json file and try building it again.

**Comment by jlevers** [comment]
I was able to fix this problem by checking out the 3.0.0 branch and rerunning `mvn clean package`.

**Comment by shrutichincholkar** [comment]
@Grandolf49 Did your "Maven Javadoc Error while building the generated code with Java 11" issue get fixed with maven-javadoc-plugin 3.2.0. I am facing the same issue with my code in Java 11 even after editing my pom.xml to maven-javadoc-plugin 3.2.0. Is there any other solution available for this issue yet?

**Comment by Grandolf49** [comment]
@shrutichincholkar Yeah, that fix worked for me. What error are you facing after editing the generated `pom.xml`?

**Comment by shrutichincholkar** [comment]
This is the current issue that i am facing. Although maven-javadoc-plugin 3.2.0 works proper for Java 12. The issue is only for Java11

[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 6.731 s
[INFO] Finished at: 2021-02-09T23:07:43+05:30
[INFO] Final Memory: 36M/134M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.2.0:jar (attach-javadocs) on project framework: MavenReportException: Error while generating Javadoc:
[ERROR] Exit code: 1 - /home/Jdk11/framework/ComponentFramework.java:28: error: cannot find symbol
[ERROR] private static FrameworkStarter frameworkStarter = null;
[ERROR] ^

**Comment by Grandolf49** [comment]
If you don't really need javadoc to be generated, a workaround would be to simply skip generating javadoc while building the project:  `mvn "-Dmaven.javadoc.skip=true" package`.

**Comment by shrutichincholkar** [comment]
Oh no I already did that just to go ahead. But we sure do need javadoc to be generated. Did you add any dependency to Javadoc plugin for 3.2.0 ?

**Comment by Grandolf49** [comment]
No, didn't require to add any other dependency.

In my case since I was getting this error: `Exit code: 1 - javadoc: error - The code being documented uses modules but the packages defined in http://docs.oracle.com/javase/8/docs/api/ are in the unnamed module.`, the `3.2.0` version fix worked for me.

I'm guessing in your case: `Exit code: 1 - /home/Jdk11/framework/ComponentFramework.java:28: error: cannot find symbol` some other fix is required. 

I'm not sure if this would help but instead of `org.hidetake:gradle-swagger-generator-plugin:2.18.2` you can try using the [OpenAPI Generator gradle plugin](https://github.com/OpenAPITools/openapi-generator/tree/master/modules/openapi-generator-gradle-plugin) to generate the client code. It's the official plugin for generating client code from openapi/swagger documentation

**Comment by shrutichincholkar** [comment]
Okay Sure need to check if it works for my code. Thanks alot for your help.
